src/dataflow/src/render/mod.rs

```diff
@@ -289,212 +240,237 @@ where
             }
         }
 
-        if let SourceConnector::External {
-            connector,
-            encoding,
-            envelope,
-            consistency,
-            max_ts_batch: _,
-            ts_frequency,
-        } = src.connector
-        {
-            let get_expr = RelationExpr::global_get(src_id, src.desc.typ().clone());
-
-            // This uid must be unique across all different instantiations of a source
-            let uid = SourceInstanceId {
-                source_id: src_id,
-                dataflow_id: self.dataflow_id,
-            };
+        match src.connector {
+            SourceConnector::Local => {
+                let ((handle, capability), stream) = scope.new_unordered_input();
+                if scope.index() == 0 {
```

Comment:
kind of an orthogonal concern but does this imply that tables are only rendered on one worker?

Reply:
Yeah, otherwise you end up with num_workers copies of the data. I reworked this to be a bit more clear. Now every worker builds the local inputs, but the inserts are only applied to worker 0.